pandas/core/common.py

```diff
@@ -2510,6 +2510,8 @@ def is_floating_dtype(arr_or_dtype):
 
 
 def is_bool_dtype(arr_or_dtype):
+    if arr_or_dtype is NotImplemented:
```

Comment:
Well, this is an internal function, so it's not extensively tested (and not worth worrying about testing too much). It expects the argument to already be a numpy dtype or numpy array.